pkg/utils/avi_rest_utils.go

```diff
@@ -38,20 +39,50 @@ func SharedAVIClients() *AviRestClientPool {
 	// TODO: Propagate error
 	ctrlUsername := os.Getenv("CTRL_USERNAME")
 	ctrlPassword := os.Getenv("CTRL_PASSWORD")
+	ctrlAuthtoken := os.Getenv("CTRL_AUTHTOKEN")
 	ctrlIpAddress := os.Getenv("CTRL_IPADDRESS")
 
-	if ctrlUsername == "" || ctrlPassword == "" || ctrlIpAddress == "" {
+	if ctrlUsername == "" || (ctrlPassword == "" && ctrlAuthtoken == "") || ctrlIpAddress == "" {
 		AviLog.Fatal(`AVI controller information missing. Update them in kubernetes secret or via environment variables.`)
 	}
 	clientonce.Do(func() {
 		AviClientInstance, _ = NewAviRestClientPool(NumWorkersGraph,
-			ctrlIpAddress, ctrlUsername, ctrlPassword)
+			ctrlIpAddress, ctrlUsername, ctrlPassword, ctrlAuthtoken)
 	})
 	return AviClientInstance
 }
 
+func NewAviRestClientWithToken(api_ep string, username string, authtoken string) *clients.AviClient {
+	var aviClient *clients.AviClient
+	var transport *http.Transport
+
+	rootPEMCerts := os.Getenv("CTRL_CA_DATA")
+	if rootPEMCerts != "" {
+		caCertPool := x509.NewCertPool()
+		caCertPool.AppendCertsFromPEM([]byte(rootPEMCerts))
+
+		transport =
+			&http.Transport{
+				TLSClientConfig: &tls.Config{
+					RootCAs: caCertPool,
+				},
+			}
+	}
+	var err error
+	if rootPEMCerts != "" {
```

Comment:
Can this be inside same if block (line no: 60) above ?

Reply:
done